Chore/55/migrate to ts

bradtaniguchi opened the pull request:
- Update all code to TS

Closes #55 

**note** this is a draft as I need to update the github action deployment to take into account the new TS flow

bradtaniguchi committed:
Add ts boilerplate

bradtaniguchi committed:
ran ts-to-js-converter script

bradtaniguchi committed:
Migrate config function files

bradtaniguchi committed:
Update part of bootstrap.js file

bradtaniguchi committed:
Migrate addFormatting function

bradtaniguchi committed:
Fix issue in index.ts for bootstrap

bradtaniguchi committed:
Update thanks, bootstrap files

bradtaniguchi committed:
Update add-formatting ts def

bradtaniguchi committed:
Update proccess-reply code

bradtaniguchi committed:
Update eightball

bradtaniguchi committed:
Update help and stats

bradtaniguchi committed:
Update nodemon and build

bradtaniguchi committed:
Update formatter code

bradtaniguchi committed:
Add commands to bootstrap

bradtaniguchi committed:
Update help command

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Update jest to support ts test files

bradtaniguchi committed:
Update package-json to support heroku deploy

bradtaniguchi committed:
Fix lint warnings in add-formatting

bradtaniguchi committed:
Update thanks.test linting

- replace all `any` calls with `never`

naomi-lgbt commented on src/config/get-config.ts:
```diff
@@ -0,0 +1,63 @@
+   * to any new user who joins the server.
+   * By default this feature is OFF
+   */
+  WELCOME_DM: boolean;
```
Quick question. This is typed as a boolean here, but compared to a string in the return. 
So my `.env` had to have it as a string and wouldn't register if I used the boolean.

bradtaniguchi replied:
All environment variables provided from `process.env` should be strings or undefined. 
So if your using `.env` then either of the following should end up as "true":
```
WELCOME_DM=true
```
or
```
WELCOME_DM='true'
```

naomi-lgbt replied:
`WELCOME_DM='true'` worked for me, but `WELCOME_DM=true` did not. 
I'll try testing it again to see if it was an error on my end.

bradtaniguchi replied:
> WELCOME_DM='true' worked for me, but WELCOME_DM=true did not.
I'll try testing it again to see if it was an error on my end.

I'll recheck with `WELCOME_DM=true`

naomi-lgbt replied:
This seems to be working now - I must have royally messed up my initial testing 😬

naomi-lgbt commented on src/commands/bootstrap.ts:
```diff
@@ -78,24 +84,22 @@ module.exports = function bootstrap({ client, config }) {
   client.on('messageDelete', function (message) {
     //change channel name to match server configuration
-    const logChannel = message.guild.channels.cache.find(
+    const logChannel = message.guild?.channels.cache.find(
```
Question mark here is breaking the deleted message logging function. 😶

bradtaniguchi replied:
This feature was working on my end, let me debug to see if I can get it to break.

What "broke" when this was ran? Did you get any errors or anything?

I'm adding an extra check around this feature to check `config.LOG_MSG_CHANNEL` is provided at all before adding this callback, so if its not available this code wont ever ran at all.

naomi-lgbt replied:
Adding the `LOG_MSG_CHANNEL` to the `.env` didn't allow the command to work until I removed the `?` and recompiled.

I'll try testing it again to see if it was an error on my part.

bradtaniguchi replied:
I'm on node version (node -v) `12.17.0` and have `tsc` version (npx tsc -v) of `3.9.5`.

@nhcarrigan Do you see an error during build (its a TS error) or when that specific code is ran (its a JS error)?

naomi-lgbt replied:
I recompiled with the `?` present and now it seems to be functioning fine. I must have made an error last time. 

Running the `npm run build` only throws the `cannot find type definition file for 'jest'` error. I'm running node version `14.2.0` and apparently only `3.9.3` of `tsc`.

bradtaniguchi committed:
Add extra check around boostrap commands

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Update coc.js to TS

bradtaniguchi committed:
Fix lint on import error

Twaha-Rahman commented on src/index.ts:
```diff
@@ -1,8 +1,8 @@
-const validateConfig = require('./config/validate-config');
-const bootstrap = require('./commands/bootstrap');
+import { Client } from 'discord.js';
+import express from 'express';
```
```suggestion
import express, { Request, Response } from 'express';
```

Then, on line 29 -

```js
    expressApp.get('/status', (_: Request, res: Response) =>
```

bradtaniguchi replied:
Explicit types are not necessary, as the `expressApp` will infer types, so the arguments are typed without us passing anything.

I'm not of fan of adding types for the sake of adding types as it increases code bloat and decreasing developer productivity, especially for those that aren't used to TS.

Twaha-Rahman replied:
Ok... got it.

Twaha-Rahman reviewed (changes requested):
I think we should be as explicit with our types as possible.

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Merge branch 'chore/55/migrate-to-ts' of github.com:bradtaniguchi/discord-bot-test into chore/55/migrate-to-ts

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

bradtaniguchi committed:
Merge with master

bradtaniguchi committed:
Merge branch 'master' into chore/55/migrate-to-ts

Commit:
Fix/suspend command (#79)

- Add bot role option, fixes issue related to the bot not being added to the suspended room

Commit:
Verbose config was incorrect (#80)

Commit:
Add forum command (#81)

- Add forum command
